cmd/sink/filters/filters.go

```diff
@@ -248,93 +248,92 @@ func (f *Filters) deleteFiltersWithMatcher(matcher NamespaceMatcher) {
 
 // Update updates the archive, delete, and archiveOnDelete filters when a ConfigMap file changes.
 func (f *Filters) Update(watcher *fsnotify.Watcher) {
-	logger := log.New(os.Stderr, "", log.LstdFlags|log.Lmicroseconds|log.LUTC)
 	for {
 		select {
 		case event, ok := <-watcher.Events:
 			if !ok { // watcher.Close() was called. We will not receive events anymore.
 				return
 			}
-			f.handleFsEvent(logger, event)
+			f.handleFsEvent(event)
 		case err, ok := <-watcher.Errors:
 			if !ok { // watcher.Close() was called. We will not receive errors anymore.
 				return
 			}
-			logger.Println("Encountered an error while watching for changes to filters:", err)
+			slog.Error("Encountered an error while watching for changes to filters", "err", err)
 		}
 	}
 }
 
 // handleFsEvent handles the logic for updating filters when an fsnotify.Event is received.
-func (f *Filters) handleFsEvent(logger *log.Logger, event fsnotify.Event) {
+func (f *Filters) handleFsEvent(event fsnotify.Event) {
 	switch event.Op {
 	case fsnotify.Create, fsnotify.Write:
 		if files.IsDirOrDne(event.Name) {
 			break
 		}
-		logger.Printf("File %s changed. Updating filters accordingly\n", event.Name)
+		slog.Info("File changed. Updating filters accordingly", "file", event.Name)
 		fileName, dir := files.FileNameAndDirFromPath(event.Name)
 		filePaths, err := files.DirectoryFiles(dir)
 		if err != nil {
-			logger.Printf("Could not get all files in directory %s: %s\n", dir, err)
+			slog.Info("Could not get all files in directory", "dir", dir, "err", err)
 			break
 		}
 		globalArchive, globalDelete, globalArchiveOnDelete, err := getGlobalCelExprs(filePaths)
 		if err != nil && !errors.Is(err, ErrNoGlobal) {
-			logger.Println("Could not read global filters:", err)
+			slog.Info("Could not read global filters", "err", err)
 			break
 		}
 		if fileName == globalKey {
-			logger.Println("Creating global filters")
+			slog.Info("Creating global filters")
 			err = f.changeGlobalFilters(filePaths, globalArchive, globalDelete, globalArchiveOnDelete)
 			if err != nil {
-				logger.Println("Problem creating global filters:", err)
+				slog.Error("Problem creating global filters", "err", err)
 			}
-			logger.Println("Successfully created global filters")
+			slog.Info("Successfully created global filters")
 			break
 		}
-		logger.Println("Creating filters for namespace:", fileName)
+		slog.Info("Creating filters for namespace", "namespace", fileName)
 		if event.Has(fsnotify.Write) {
 			err = f.updateNamespaceFilters(fileName, event.Name, globalArchive, globalDelete, globalArchiveOnDelete)
 		} else {
 			err = f.createNamespaceFilters(fileName, event.Name, globalArchive, globalDelete, globalArchiveOnDelete)
 		}
 		if err != nil {
-			logger.Printf("Problem creating filters for namespace %s: %s\n", fileName, err)
+			slog.Error("Problem creating filters for namespace", "namespace", fileName, "err", err)
```

Comment:
Shouldn't you have an exit here? If you don't, you'll have an error message "Problem creating filters for namespace" immediately followed by an info message "Created filters for namespace". If you are going to keep processing even with this failure, I suggest you (1) make this message a warning (`slog.Warn`) and (2) put the info message in an else.

Reply:
A warning will slip easier as something we don't need to pay attention to, so I think its better to have it as an error. About the logging statements themselves... @skoved